src/Modal.js

```diff
@@ -55,7 +55,7 @@ const Modal = React.createClass({
         className={classNames(this.props.className, classes)}
         onClick={this.props.backdrop === true ? this.handleBackdropClick : null}
         ref="modal">
-        <div className={classNames(dialogClasses)}>
+        <div className={classNames(this.props.dialogClassName, dialogClasses)}>
```

Comment:
Anyway I've checked it and I understand what the issue is now.

```js
let modal = (
  <div
...
return this.props.backdrop ? 
  this.renderBackdrop(modal) : modal; // here is the cause. 
```
When `return` returns something other than outer element (e.g. expression, variable or constant),
then `elint-plugin-react` linter has no clue how to check props usage and type checks.
It doesn't perform the full code evaluation.